faucet/src/http.rs

```diff
@@ -196,38 +199,32 @@ async fn atomic_faucet_funding(
     parachain_rpc: &InterBtcParachain,
     kv: Bucket<'_, String, Json<FaucetRequest>>,
     account_id: AccountId,
-    currency_id: CurrencyId,
-    allowances: HashMap<FundingRequestAccountType, u128>,
+    allowances: HashMap<FundingRequestAccountType, &Allowance>,
 ) -> Result<(), Error> {
-    let account_str = format!("{}-{}", account_id, currency_id.inner()?.symbol());
+    let account_str = format!("{}", account_id);
     let last_request_json = kv.get(account_str.clone())?;
     let account_type = get_account_type(parachain_rpc, account_id.clone()).await?;
+    let amounts = allowances.get(&account_type).ok_or(Error::NoFaucetAllowance)?;
+    let currency_ids: Vec<CurrencyId> = amounts.iter().map(|(currency, _)| *currency).collect();
     ensure_funding_allowed(
         parachain_rpc,
         account_id.clone(),
-        currency_id,
+        currency_ids,
         last_request_json,
         account_type.clone(),
     )
     .await?;
 
-    let amount = allowances
-        .get(&account_type)
-        .ok_or(Error::NoFaucetAllowance)?
-        .checked_mul(currency_id.inner()?.one())
-        .ok_or(Error::MathError)?;
-
-    log::info!(
-        "AccountId: {}, Currency: {:?} Type: {:?}, Amount: {}",
-        account_id,
-        currency_id.inner()?.symbol(),
-        account_type,
-        amount
-    );
-
-    let mut transfers = vec![parachain_rpc.transfer_to(&account_id, amount, currency_id)];
-    if currency_id != parachain_rpc.native_currency_id {
-        transfers.push(parachain_rpc.transfer_to(&account_id, amount, parachain_rpc.native_currency_id));
+    let mut transfers = vec![];
+    for (currency_id, amount) in amounts.iter() {
+        log::info!(
+            "AccountId: {}, Currency: {:?} Type: {:?}, Amount: {}",
+            account_id,
+            currency_id.inner()?.symbol(),
```

Comment:
This will fail on foreign currencies: https://github.com/interlay/interbtc-clients/blob/d5af9c500e1739675bd4d960828cecb5fa6f5fd6/runtime/src/types.rs#L151